src/Screens/MainPage/components/SearchMainPage.tsx

```diff
@@ -3,62 +3,91 @@ import Search from "../../../Comps/search/Search";
 import useDebounce from "../../../Hooks/useDebounce";
 import { SearchIcon } from "../../../Icons";
 import type { RootState } from "../../../store/store";
-
-import { useSelector, useDispatch } from "react-redux";
-import { updateSearch } from "../../../store/query";
-import {FormControl } from "@mui/material";
 import { VerticalLine } from "../../../Comps/search/style";
 import DropDown from "../../../Comps/select/DropDown";
-import {Container} from './style'
+import { Container } from "./style";
+import {
+  updateEndPoint,
+  updateQueryUrl,
+  updateSearch,
+} from "../../../store/query";
+import { BASE_URL, API_KEY } from "../../../Services/Api";
 
+import { useSelector, useDispatch } from "react-redux";
+import { FormControl } from "@mui/material";
 
 export const SearchMainPage = () => {
   const [input, setInput] = useState<string>("");
+  const Query = useSelector((state: RootState) => state.query);
+
   const debouncedValue = useDebounce<string>(input, 500);
   const placeholders = ["Top Headlines", "Everything"];
 
-  const Query = useSelector((state: RootState) => state.query);
   const dispatch = useDispatch();
 
   const handleChange = (value: string) => {
     setInput(value);
-    dispatch(updateSearch({ input: value}));
-    console.log(Query.query);
   };
-  const handleDropDown = (value: string) => {
-    console.log(value);
-
+  const handleEndpointDropDown = (value: string) => {
+    if (value === "Everything") dispatch(updateEndPoint("everything"));
+    else {
+      dispatch(updateEndPoint("top-headlines"));
+    }
+  };
+  const isQueryEmpty = () => {
+    let isEmpty;
+    Query.query.filters.sources.length === 0
+      ? (isEmpty = true)
+      : (isEmpty = false);
+    return isEmpty;
   };
-
   useEffect(() => {
-    //change input in store redux
+    dispatch(updateSearch(input));
   }, [debouncedValue]);
 
+  const buildApiQuery = () => {
+    let queryUrl = "";
+    if (Query.query.endpoint === "top-headlines") {
+      if (isQueryEmpty()) {
+        queryUrl = `${BASE_URL}${Query.query.endpoint}?country=${Query.query.filters.country}&category=${Query.query.filters.category}&apiKey=${API_KEY}`;
+      } else {
+        queryUrl = `${BASE_URL}${Query.query.endpoint}?sources=${Query.query.filters.sources}&category=${Query.query.filters.category}&apiKey=${API_KEY}`;
+      }
+    } else {
+      queryUrl = `${BASE_URL}${Query.query.endpoint}?q=${Query.query.search}&sortBy=${Query.query.sortby}&from=${Query.query.filters.date}&to=${Query.query.filters.date}&sources=${Query.query.filters.sources}&language=${Query.query.filters.language}&apiKey=${API_KEY}`;
+    }
+    dispatch(updateQueryUrl(queryUrl));
+  };
   useEffect(() => {
-    //call api
-  }, [Query]);
+    buildApiQuery();
+  }, [
+    Query.query.filters,
```

Comment:
deconstruct that object somewhere in the file to:
```suggestion
    {filters, sortby, search, endpoint} = Query.query
```

Reply:
that way you don't need to write Query.query. five times